### DISCUSSION TITLE
Private emails

### DISCUSSION THREAD

**Discussion by anirudhb** [npm/feedback#189] [Registry] [answered]
I don't really want to give out my personal email address when publishing packages. I would prefer if I could use GitHub's anonymous email as a "public" facing email, and then have a personal email used for the sole purpose of verifying my account.

**Comment by MylesBorins** [comment, marked as answer]
@anirudhb thank you for making this suggestion. At this time we are not going to be able to make changes such as supporting GitHub's anonymous emails, but we will keep this in mind as we explore new features such as signing into npm with GitHub Auth.

**Comment by dstaley** [comment]
I would love to second this idea. I think it's honestly quite bizarre that I have to provide my email address when publishing a package to npm, and that there's no way to separate the email linked to a published package from my npm account email. Since those two have to be the same, I'm unable to even do something like filter out emails sent to my npm address since I'd miss important emails from npm.

**Comment by simarist** [comment]
it is frustrating that you do not have a choice but to make your email completely public, you could so easily make a bot to get pretty much all the emails of npm users...

**Comment by Purpzie** [comment]
Until NPM adds support for this themselves, [firefox relay](https://relay.firefox.com) is a good alternative. I'm currently using it.

**Comment by trekzavier** [comment]
Yeah, please add this. I searched for this specific feature to upvote it because I just experienced something rather unsettling. 

All of a sudden, these past few weeks, I've just started getting all of these recruiters spamming my inbox with their job offers.  My npm email is my personal email that I never use for job applications, so I found it very, very odd to receive job related emails at this particular email address.

Out of curiosity, I asked one of them how they were able to retrieve my email address.  He quite candidly explained that they were running `npm view` on my packages and found my npm email there. 

Frankly, for someone who is extremely privacy conscious, this is unacceptable. My npm email (which was meant to stay private) has now been placed on some recruiting lists and potentially others and I'm receiving spam now because of npm. Not good.
